modules/cudaoptflow/include/opencv2/cudaoptflow.hpp

```diff
@@ -43,6 +43,9 @@
 #ifndef OPENCV_CUDAOPTFLOW_HPP
 #define OPENCV_CUDAOPTFLOW_HPP
 
+#include "NVIDIAOpticalFlowSDK_1_0_Headers/nvOpticalFlowCommon.h"
+#include "NVIDIAOpticalFlowSDK_1_0_Headers/nvOpticalFlowCuda.h"
```

Comment:
Do we really need that headers as a part of public OpenCV API?

Usually we hide all implementation details. In general, inclusion of OpenCV header should not affect names outside of "cv::" namespace or `CV_` macros/defines
For enums we "duplicate" in corresponding classes (at least such duplication is needed for bindings generator).

Reply:
We will need those headers since all the enums and function pointers are declared there and every release of new SDK will have some parameters/enums/functions added. So yes, headers are must.

Reply:
You need headers, but exposing them as public OpenCV API is not necessary.

If you add "duplicate" of enum `NV_OF_PERF_LEVEL` here (renamed, but keep the same values, so just cast is enough), then you can use headers from .cpp files only (move headers into `src/` subtree) and don't expose them for users.

Reply:
True. Probably we can do it but again once i may need to discuss it within my team. Il move the headers and update the CL  once confirmed.

Reply:
I moved headers to src and duplicated NV_OF_PERF_LEVEL as NVIDIA_OF_PERF_LEVEL